graylog2-web-interface/src/preflight/hooks/useRenewalPolicy.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+
+import type { RenewalPolicy } from 'preflight/types';
+import type FetchError from 'logic/errors/FetchError';
+import fetch from 'logic/rest/FetchProvider';
+import { qualifyUrl } from 'util/URLUtils';
+
+export const QUERY_KEY = ['data-nodes', 'renewal-policy'];
+const fetchRenewalPolicy = (): Promise<RenewalPolicy> => (
+  fetch('GET', qualifyUrl('/api/renewal_policy'), undefined, false)
+);
+
+const useRenewalPolicy = (): {
+  data: RenewalPolicy,
+  isFetching: boolean,
+  error: FetchError,
+  isInitialLoading: boolean
+} => {
+  const {
+    data,
+    isFetching,
+    error,
+    isInitialLoading,
+  } = useQuery<RenewalPolicy, FetchError>({
+    queryKey: QUERY_KEY,
+    queryFn: fetchRenewalPolicy,
+    initialData: undefined,
+    retry: 3000,
+  });
+
+  return { data, isFetching, error, isInitialLoading };
+};
+
+export default useRenewalPolicy;
```

Comment:
It would be nice to have a test for the `useRenewalPolicy` hook.

Reply:
I think its value would be very slim and effectively testing `react-query`, as `useRenewalPolicy` is not doing much at all right now.